src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/PrisonApiService.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.service
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import uk.gov.justice.digital.hmpps.visitscheduler.client.PrisonApiClient
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.OffenderNonAssociationDetailDto
+
+@Service
+class PrisonApiService(
+  private val prisonApiClient: PrisonApiClient
+) {
+
+  companion object {
+    val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getOffenderNonAssociationList(prisonerId: String): List<OffenderNonAssociationDetailDto> {
+    try {
+      val offenderNonAssociationList = prisonApiClient.getOffenderNonAssociation(prisonerId)?.nonAssociations ?: emptyList()
+      SessionService.LOG.debug("sessionHasNonAssociation prisonerId : $prisonerId has ${offenderNonAssociationList.size} non associations!")
+      return offenderNonAssociationList
+    } catch (e: WebClientResponseException) {
+      if (e.statusCode != HttpStatus.NOT_FOUND) {
+        LOG.error("Exception thrown from call to prison API : $e")
+        throw e
```

Comment:
why just NOT_FOUND, would it be useful to throw exception  in all exception cases!

Reply:
I think this is because prison API uses http error codes incorrectly, when it cant find prison it return a 404. So after all this is needed. It should not do this it should return a 200 when it cant find something,,... no result status